### PR TITLE
Improve some tests

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Cleans up a few tests.

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [n/a] Did you make sure to update the documentation with your changes?
- [n/a] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._

### CODE REVIEW

File: tests/checkpointing/test_model_checkpoint.py [hunk 1]
```diff
@@ -663,92 +658,30 @@ def validation_step(self, batch, batch_idx):
...
-            enable_pl_optimizer=enable_pl_optimizer)
-        trainer.fit(model)
-        trainer.test(model)
-
-        assert str(os.listdir(tmpdir)) == "['epoch=00.ckpt']"
-
-
-@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
-@pytest.mark.parametrize("enable_pl_optimizer", [False, True])
-def test_checkpoint_repeated_strategy_tmpdir(enable_pl_optimizer, tmpdir):
```
Comment (carmocca) [inline tests/checkpointing/test_model_checkpoint.py:696]: Note for reviewers: I merged these two tests because they are very similar and we don't need to test the `filepath` parameter here, we have tests for it already and it is deprecated

File: tests/trainer/test_trainer.py [hunk 2]
```diff
@@ -507,180 +505,144 @@ def test_resume_from_checkpoint_epoch_restored(monkeypatch, tmpdir, tmpdir_serve
...
+        progress_bar_refresh_rate=0,
     )
-
-    # fit model
-    trainer = Trainer(**trainer_options)
     result = trainer.fit(model)
-    assert result == 1, "Training did not complete"
 
-    # check training stopped at max_steps
+    assert result == 1, "Training did not complete"
```
Comment (Borda) [inline tests/trainer/test_trainer.py:645]: just thinking about rather move to test the trainer state if it finished...

**Review by Borda — APPROVED** [review]
nice one :]

File: tests/trainer/test_trainer.py (hunk 2 above)
Comment (carmocca) [reply tests/trainer/test_trainer.py:645]: Could test both the result and the trainer state if you want

Comment (Borda) [reply tests/trainer/test_trainer.py:645]: I am thinking that we shall deprecate this returning 1, it does make much sense...

Comment (carmocca) [reply tests/trainer/test_trainer.py:654]: Added state asserts :]

**Review by tchaton — APPROVED** [review]
LGTM ! Nice cleanup.

**Review by s-rog — APPROVED** [review]
